### PR TITLE
REF: Keep Document() simple; add Document.from_mongo.

### PR DESCRIPTION
**Pull request by danielballan**

### CODE REVIEW

File: metadatastore/document.py [hunk 1]
```diff
@@ -105,3 +81,33 @@ def __len__(self):
...
+    Copy the data out of a mongoengine.Document, including nested Documents,
+    but do not copy any of the mongo-specific methods or attributes.
+    """
+    def __init__(self, mongo_document):
+        """
+        Parameters
+        ----------
+        mongo_document : mongoengine.Document
+        """
+        self._fields = set()
```
Comment (tacaswell) [inline metadatastore/document.py]: call `super` here instead?

Comment (danielballan) [inline metadatastore/document.py]: Done.

**Comment by danielballan** [comment]
Added to this PR: Log messages used to refer to mongo IDs. Now they refer to our globally unique uid.

**Comment by danielballan** [comment]
The intention of this PR has changed (see updated title).

File: metadatastore/document.py [hunk 2]
```diff
@@ -43,30 +43,10 @@ def _normalize(in_val):
...
-        """
-        Parameters
-        ----------
-        mongo_document : mongoengine.Document
-        """
-        self._fields = set()
-        self._name = mongo_document.__class__.__name__
-        fields = set(chain(mongo_document._fields.keys(),
-                           mongo_document._data.keys()))
+    """A dictionary where d.key is the same as d['key']"""
```
Comment (tacaswell) [inline metadatastore/document.py]: Include a note about the magic with `_` prefixed attributes.

Comment (tacaswell) [inline metadatastore/document.py]: as related to iteration.

File: metadatastore/document.py [hunk 3]
```diff
@@ -105,3 +81,34 @@ def __len__(self):
...
+        document._name = mongo_document.__class__.__name__
+        fields = set(chain(mongo_document._fields.keys(),
+                           mongo_document._data.keys()))
+
+        for field in fields:
+            attr = getattr(mongo_document, field)
+
+            attr = _normalize(attr)
+
+            setattr(document, field, attr)
```
Comment (tacaswell) [inline metadatastore/document.py]: `document[field] = attr`

File: metadatastore/document.py [hunk 4]
```diff
@@ -105,3 +81,34 @@ def __len__(self):
...
+            # For debugging, add a human-friendly time_as_datetime attribute.
+            if hasattr(document, 'time'):
```
Comment (tacaswell) [inline metadatastore/document.py]: `if 'time' in document:`

**Pull request merged**